web/form.py

```diff
@@ -143,7 +149,10 @@ def is_hidden(self):
         return False
         
     def get_type(self):
-        raise NotImplementedError()
+        if self.type != None:
+            return self.type
+        else:
+            raise AttributeError("missing attribute 'type'")
```

Comment:
Why not just `return self.type`?  (would also raise AttributeError).
Anyway: `!= None` should be `is not None`.

Reply:
But in general I think
```python
try:
    return self.type
except AttributeError:
    return None
```
would be best.